### PR TITLE
Make sure raw URIs are wellformed

### PR DESCRIPTION
**Pull request by tomkerkhove**
Make sure raw URIs are wellformed.

Relates to #67

### CODE REVIEW

**Comment by tomkerkhove** [comment]
@stijnmoreels 	The `ForTopic_NonHttpEndpointTopic_ShouldFailWithUriFormatException` test will fail on `file:///C:\\temp\\dir` saying it is not well formed.

Is that actually the case? Where did you get this test value from?

**Comment by stijnmoreels** [comment]
> @stijnmoreels The `ForTopic_NonHttpEndpointTopic_ShouldFailWithUriFormatException` test will fail on `file:///C:\\temp\\dir` saying it is not well formed.
> 
> Is that actually the case? Where did you get this test value from?

This was was made to not test for invalid URI's in the context of invalid strings but in the context of valid URI endpoints that are not HTTP(S).

File: src/Arcus.EventGrid.Publishing/EventGridPublisherBuilder.cs [hunk 1]
```diff
@@ -57,6 +57,8 @@ private EventGridPublisherBuilder(Uri topicEndpoint)
         public static EventGridPublisherBuilder ForTopic(string topicEndpoint)
         {
             Guard.NotNullOrWhitespace(topicEndpoint, nameof(topicEndpoint), "The topic endpoint must not be empty and is required");
+            Guard.For<ArgumentException>(() => Uri.IsWellFormedUriString(topicEndpoint, UriKind.Absolute) == false, "The topic endpoint is not a valid URI");
```
Comment (stijnmoreels) [inline src/Arcus.EventGrid.Publishing/EventGridPublisherBuilder.cs]: If this could be used, it would indeed be an improvement!

**Comment by tomkerkhove** [comment]
> > @stijnmoreels The `ForTopic_NonHttpEndpointTopic_ShouldFailWithUriFormatException` test will fail on `file:///C:\\temp\\dir` saying it is not well formed.
> > Is that actually the case? Where did you get this test value from?
> 
> This was was made to not test for invalid URI's in the context of invalid strings but in the context of valid URI endpoints that are not HTTP(S).

Is it correct to say you came up with it or is it a real value? Given the test is failing I'm not sure if it's because of the functionality or the test value.

**Comment by stijnmoreels** [comment]
> > > @stijnmoreels The `ForTopic_NonHttpEndpointTopic_ShouldFailWithUriFormatException` test will fail on `file:///C:\\temp\\dir` saying it is not well formed.
> > > Is that actually the case? Where did you get this test value from?
> > 
> > 
> > This was was made to not test for invalid URI's in the context of invalid strings but in the context of valid URI endpoints that are not HTTP(S).
> 
> Is it correct to say you came up with it or is it a real value? Given the test is failing I'm not sure if it's because of the functionality or the test value.

Both actually. I came up with this real test value.

**Comment by tomkerkhove** [comment]
That's odd given it doesn't really work - Not sure why it doesn't work then..

**Pull request merged**